pkg/metrics/bpf.go

```diff
@@ -36,55 +52,70 @@ func newbpfCollector() *bpfCollector {
 }
 
 func (s *bpfCollector) Describe(ch chan<- *prometheus.Desc) {
-	ch <- s.bpfMapsMemory
-	ch <- s.bpfProgMemory
+	prometheus.DescribeByCollect(s, ch)
 }
 
 type memoryEntry struct {
 	BytesMemlock uint64 `json:"bytes_memlock"`
 }
 
-func getMemoryUsage(typ string) (uint64, error) {
+func getBPFUsage(typ string) (uint64, uint64, error) {
```

Comment:
Might be a good idea to return a struct instead of the 2 `uint64`, to make it more obvious what each value corresponds to.

Reply:
sounds good indeed, I've added a new `bpfUsage` struct to support those variables.